SLHCUpgradeSimulations/L1TrackTrigger/plugins/L1TkMuonFromExtendedProducer.cc

```diff
@@ -140,137 +179,188 @@ L1TkMuonFromExtendedProducer::produce(edm::Event& iEvent, const edm::EventSetup&
   iEvent.getByLabel(L1TrackInputTag_, l1tksH);
   const L1TkTrackCollectionType& l1tks = *l1tksH.product();
 
+  LogDebug("L1TkMuonFromExtendedProducer") << " l1mus.size= " << l1mus.size() << " l1tks.size= " << l1tks.size();
+
   L1TkMuonParticleCollection l1tkmuCands;
   l1tkmuCands.reserve(l1mus.size()*4); //can do more if really needed
 
+  // MuCn filling
+
   int imu = -1;
+  int MuonNum = -1;
   for (auto l1mu : l1mus){
     imu++;
+
+    if( imu<MuonsMax ) {
+      MuonNum = imu + 1;
+      MuCn[imu].eta = l1mu.eta();
+      MuCn[imu].eta = l1mu.eta();
+      MuCn[imu].phi = l1mu.phi();
+      MuCn[imu].pt = l1mu.pt();
+      MuCn[imu].bx = l1mu.bx();
+      MuCn[imu].quality = l1mu.quality();
+      MuCn[imu].sigmaEta = l1mu.sigmaEta();
+      MuCn[imu].sigmaPhi = l1mu.sigmaPhi();
+      MuCn[imu].feta = fabs( l1mu.eta());
+  
+      const auto& gmtCand = l1mu.gmtMuonCand();
+      const auto& dtCand  = l1mu.dtCand();
+      const auto& cscCand = l1mu.cscCand();
+      const auto& rpcCand = l1mu.rpcCand();
+  
+      MuCn[imu].gmtCand = gmtCand.empty();
+      MuCn[imu].dtCand = dtCand.empty();
+      MuCn[imu].cscCand = cscCand.empty();
+      MuCn[imu].rpcCand = rpcCand.empty();
+  
+      LogDebug("L1TkMuonFromExtendedProducer") << " imu= " << imu
+        << " eta= " << MuCn[imu].eta 
+        << " phi= " << MuCn[imu].phi 
+        << " bx= " << MuCn[imu].bx
+        << " quality= " << MuCn[imu].quality
+        << " sigmaEta= " << MuCn[imu].sigmaEta
+        << " sigmaPhi= " << MuCn[imu].sigmaPhi 
+        << " gmtCand " << MuCn[imu].gmtCand 
+        << " dtCand " << MuCn[imu].dtCand 
+        << " cscCand " << MuCn[imu].cscCand 
+        << " rpcCand " << MuCn[imu].rpcCand;
+    } 
+    else { 
+      LogWarning("L1TkMuonFromExtendedProducer") << " Maximum Muon number too small " << imu;
+    };
+
+    // trCn filling
+  
+    PropState matchProp;
+    float drmin = 999;
+    int TrackNume = -1;
+    int il1tk = -1;
+    for (auto l1tk : l1tks ){
+      il1tk++;
+  
+      if( imu<MuonsMax ) {
+
+        unsigned int nPars = 4;
+        if (use5ParameterFit_) nPars = 5;
+        L1Tk[imu].nPars = nPars;
```

Comment:
values in ```  if (dr2prop < drmin){ ``` will be updated only if this track is the closest, while all other L1Tk values will come from the last track in the list